### PR TITLE
Add JJB Job test for Jenkins RPC PRs

### PR DESCRIPTION
**Pull request by hughsaunders**
Adds a job that tests changes to JJB job definitions.
This is a simple test to ensure that proposed changes to compile.

Modifies the run_jjb script to add an operation paramter, this allows
the script to be used for checks as well as for syncing repo definitions
with jenkins.

Connects rcbops/u-suk-dev#345
Depends (includes) rcbops/jenkins-rpc#639

### CODE REVIEW

File: rpc-jobs/jobs.yaml [hunk 1]
```diff
@@ -18,22 +23,311 @@
...
+
+            # Collect MAAS agent config
+            cp -r /etc/rackspace-monitoring-agent.conf.d archive/etc/
+            cp /etc/rackspace-monitoring-agent.cfg archive/etc/
+
+            # Jenkins user must be able to read all files to archive and transfer to
+            # the master. Unreadable files will cause the whole archive operation to fail
+            chown -R jenkins archive
+
+            # remove dangling/circular symlinks
```
Comment (odyssey4me) [inline rpc-jobs/jobs.yaml]: Those pesky dangling symlinks!

Comment (andymcc) [inline rpc-jobs/jobs.yaml]: They're the worst.

**Comment by odyssey4me** [comment]
:+1: Yeah, this LGTM

**Comment by andymcc** [comment]
👍

**Pull request merged**